misk-zookeeper/src/main/kotlin/misk/zookeeper/CuratorFrameworkProvider.kt

```diff
@@ -1,15 +1,36 @@
-package misk.clustering.zookeeper
+package misk.zookeeper
 
 import com.google.common.util.concurrent.ThreadFactoryBuilder
+import misk.clustering.zookeeper.ZookeeperConfig
+import misk.clustering.zookeeper.asZkPath
 import org.apache.curator.framework.CuratorFramework
 import org.apache.curator.framework.CuratorFrameworkFactory
+import org.apache.curator.framework.api.ACLProvider
 import org.apache.curator.retry.ExponentialBackoffRetry
-import org.apache.curator.utils.DefaultZookeeperFactory
+import org.apache.zookeeper.ZooDefs
+import org.apache.zookeeper.ZooDefs.Ids.ANYONE_ID_UNSAFE
+import org.apache.zookeeper.ZooDefs.Ids.AUTH_IDS
 import org.apache.zookeeper.ZooKeeper
 import org.apache.zookeeper.client.ZKClientConfig
+import org.apache.zookeeper.data.ACL
+import java.util.Collections
 import javax.inject.Inject
 import javax.inject.Provider
 
+// Directory where ZooKeeper stores the leases that it manages on a service's behalf
+const val LEASES_NODE = "leases"
```

Comment:
what if a service wants to use zk for more than just leases?

Reply:
Then it can use the `ZkClientFactory` and put whatever it wants in `/services/<app-name>/`. The client factory will give them a client configured to do just that.

Reply:
so why the need for leases and services ?

Reply:
hmm because leases are a different namespace...

why aren't leases scoped to the service?

Reply:
I don't think there's a good reason, tbh. I think that's a relic of the fact that the LeaseManager code predated the ZK work. I think we can move that into the `/services` namespace with little problem unless there's something I'm missing. If we want to do that, it's better to do it now rather than later.

Reply:
my vote is to move it, but i'm not familiar with the problem space.

would it simplify the ACL logic if everything was nested under `/services`?

Reply:
i don't know enough about zk to have an opinion.

Reply:
There's nothing really special around these paths. They're just namespaces so we can set them up however we see fit (highly analogous to organize files on disk)

I'm going to go with `/services/<app-name>/leases` and `/services/<app-name>/data` since it seems to be the most logical to me.

Reply:
is it safe to rollout this change? is it going to break clustering during the rollout? probably not a big deal since nobody in prod depends on clustering yet

Reply:
Yeah, exactly. Plus, I was planning on blowing away ZK anyway when I do this so that we can pick up the encrypted volumes changes.